src/transformers/modeling_utils.py

```diff
@@ -2490,8 +2490,7 @@ def save_pretrained(
                 save_function(shard, os.path.join(save_directory, shard_file))
 
         if index is None:
-            weights_file_name = SAFE_WEIGHTS_NAME if safe_serialization else WEIGHTS_NAME
-            path_to_weights = os.path.join(save_directory, _add_variant(weights_file_name, variant))
+            path_to_weights = os.path.join(save_directory, _add_variant(weights_name, variant))
```

Comment:
Following from L2457 - L2459, do we still want use `_add_variant` when weights_name is an adapter name like `ADAPTER_SAFE_WEIGHTS_NAME`?

Reply:
Good point. I think we can definitely omit _add_variant here. If we want to use `_add_variant` with `ADAPTER_SAFE_WEIGHTS_NAME`, we should place it above these lines and, ideally, in a separate task.